cairo-libs/packages/cometbft/src/utils.cairo

```diff
@@ -33,3 +33,69 @@ impl FractionAsProtoName of ProtoName<Fraction> {
         "Fraction"
     }
 }
+
+pub impl SpanU8TryIntoU256 of TryInto<Span<u8>, u256> {
+    /// Decodes as big endian.
+    fn try_into(self: Span<u8>) -> Option<u256> {
+        // Only allows size 32 to ensure all bytes fit exactly into a `u256`.
+        if (self.len() != 32) {
+            return Option::None;
+        }
+
+        // Precomputed powers to remove the overhead of repeated multiplications.
+        let mut ret: u256 = 0;
+        let two_pow_0 = 0x1;
+        let two_pow_1 = 0x100;
+        let two_pow_2 = 0x10000;
+        let two_pow_3 = 0x1000000;
+        let two_pow_4 = 0x100000000;
+        let two_pow_5 = 0x10000000000;
+        let two_pow_6 = 0x1000000000000;
+        let two_pow_7 = 0x100000000000000;
+        let two_pow_8 = 0x10000000000000000;
+        let two_pow_9 = 0x1000000000000000000;
+        let two_pow_10 = 0x100000000000000000000;
+        let two_pow_11 = 0x10000000000000000000000;
+        let two_pow_12 = 0x1000000000000000000000000;
+        let two_pow_13 = 0x100000000000000000000000000;
+        let two_pow_14 = 0x10000000000000000000000000000;
+        let two_pow_15 = 0x1000000000000000000000000000000;
```

Comment:
I think, the variable names are bit misleading. it's `n256_pow_0`, `n256_pow_1` and the rest. I will let you decide the naming.

also, you may declare `ret` after this block.